pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -704,6 +707,47 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 	return nil
 }
 
+func (r *Reconciler) handleConcurrencyControls(ctx context.Context, pr *v1beta1.PipelineRun) error {
+	concurrencyRef, ok := pr.Labels["tekton.dev/concurrency"]
+	if !ok {
+		return nil
+	}
+	_, ok = pr.Labels["tekton.dev/concurrency-key"]
+	if ok {
+		// Assume concurrency key was added in a previous reconcile loop, and other pipelineRuns have already been canceled.
+		return nil
+	}
+	cc, err := r.concurrencyControlLister.ConcurrencyControls(pr.Namespace).Get(concurrencyRef)
+	if err != nil {
+		return fmt.Errorf("error getting concurrency control for PR %s: %s", pr.Name, err)
+	}
+	key := resources.ApplyParametersToConcurrency(ctx, cc, pr)
+	pr.Labels["tekton.dev/concurrency-key"] = key
+	if cc.Spec.Strategy != "Cancel" {
```

Comment:
I'm assuming we'd create constants for supported strategies?

Reply:
definitely!

Reply:
Just making sure. =)